packages/orm/src/relations/AbstractPropertyImpl.ts

```diff
@@ -0,0 +1,12 @@
+/** A base class for relations & non-relation (i.e. `PersistedAsyncProperty) properties. */
+export abstract class AbstractPropertyImpl<T> {
+  #entity: T;
+
+  constructor(entity: T) {
```

Comment:
question: what does using a symbol based field give us? could we just do 

```suggestion
  constructor(public readonly entity: T) {
```

Reply:
Yeah, I'm like 70% sure that `public readonly entity` becomes as owned field, so shows up in `Object.keys` / etc., but that `#entity + getter` w/the getter defined on the class, won't show up.

Yeah:

```
> class Foo {
... a = 1
... #b = 2
... get b() { return this.#b; }
... }
undefined
> f = new Foo();
Foo { a: 1 }
> Object.keys(f);
[ 'a' ]
>
```

At some point, this was important to avoid Jest and/or JSON.stringify (probably Jest) crawling into "all the things" when doing diffs of failures/etc.

Not entirely sure how necessary it still is...